### PR TITLE
feat(client): add file preview for images, video, audio, PDF, and text-like files

### LINKED ISSUE #134: Preview files
For selected file types, add a way to preview them. Currently users can preview text messages. We should expand this support for selected file types.

### Tier 1 — Browser-native rendering
- **Images** (`.jpg`, `.jpeg`, `.png`, `.gif`, `.webp`, `.svg`, `.bmp`) — `` in a modal
- **Video** (`.mp4`, `.webm`) — `<video>` element
- **Audio** (`.mp3`, `.wav`, `.ogg`, `.m4a`) — `<audio>` element
- **PDF** (`.pdf`) — `<iframe>` or `<object>` embed

### Tier 2 — Text-like files (render like text items)
- **Plain text** (`.txt`, `.log`, `.csv`, `.ini`, `.conf`)
- **Structured data** (`.json`, `.xml`, `.yaml`, `.yml`, `.toml`)
- **Markdown** (`.md`) — could render formatted or raw
- **Code** (`.js`, `.ts`, `.py`, `.java`, `.cs`, `.go`, etc.) — optionally with syntax highlighting

### Tier 3 — Skip (no browser-native preview)
- Documents (`.docx`, `.odt`, `.rtf`) — need conversion libraries
- Spreadsheets (`.xlsx`, `.ods`) — same
- Archives (`.zip`, `.tar`, `.gz`, `.7z`)
- Binaries

## Implementation Notes

- **No server changes needed for Tier 1** — the client already has the download URL and filename; fetch the blob and render in a modal using the appropriate HTML element
- **Tier 2** — fetch as text and display in the existing text modal (with `whitespace-pre-wrap`)
- File type detection is already done client-side in `file-icons.ts` — reuse that extension mapping
- Consider a **file size cap** for preview (e.g. skip preview for images >10 MB or text >1 MB)
- The download endpoint returns `application/octet-stream` hardcoded — that's fine since the client can create object URLs from the blob regardless

### PR DESCRIPTION
**Pull request by maraf**
Add click-to-preview support for file items in the space view. Previously only text items had a preview  file items could only be downloaded.modal 

## Tier  Browser-native preview1 
- **Images** (`.jpg`, `.jpeg`, `.png`, `.gif`, `.webp`, `.svg`, `.bmp`, `. `` in a modalico`) 
- **Video** (`.mp4`, `. `<video>` with controlswebm`) 
- **Audio** (`.mp3`, `.wav`, `.ogg`, `.m4a`, `.flac`, `. `<audio>` with controlsaac`) 
- **PDF** (`. `<iframe>` embedpdf`) 

## Tier  Text-like file preview2 
- **Plain text** (`.txt`, `.log`, `.csv`, `.ini`, `.conf`)
- **Structured data** (`.json`, `.xml`, `.yaml`, `.yml`, `.toml`)
- **Markdown** (`.md`) and **code files** (`.js`, `.ts`, `.py`, `.java`, `.cs`, `.go`, etc.)

Fetched as text from the download endpoint and displayed in the existing text modal style (`whitespace-pre-wrap`).

## Implementation details
- New `file-preview.ts` module with `getFilePreviewType()`, `isPreviewable()`, and `isFileTooLargeForPreview()` helpers
- Size guards per type (10 MB images, 100 MB video, 50 MB audio, 20 MB PDF, 1 MB text)
- Loading spinner while fetching, error fallback with download button
- Object URL cleanup on modal close
- Non-previewable files (`.zip`, `.docx`, `.xlsx`, etc.) keep download-only behavior

## Tests
- 80 unit tests for ` all preview categories, case insensitivity, edge cases (no extension, double extensions, hidden files)getPreviewType()` 
- 29 UI component  click-to-open, per-type rendering, loading state, error handling, size guard, modal closetests 

Closes #134

### CODE REVIEW

File: src/SharedSpaces.Client/e2e/screenshots.spec.ts [hunk 1]
```diff
@@ -226,6 +263,44 @@ test.describe('Screenshot Capture', () => {
...
+      await injectTokens(page, tokenMap);
+      await page.reload();
+      await page.waitForSelector('app-shell');
+      await page.click('nav button:first-child');
+      await page.waitForSelector('space-view');
+      await page.waitForTimeout(1000);
+
+      // Click the image file item to open preview
+      await page.locator('p.cursor-pointer:has-text("photo.png")').click();
+      await page.waitForSelector('button[aria-label="Close preview"]', { timeout: 5_000 });
```
[the remark this answers is not in the document]
Comment (maraf) [reply src/SharedSpaces.Client/e2e/screenshots.spec.ts]: ✅ Fixed in `1bc2def`. Updated to `page.getByRole('button', { name: 'photo.png' })` — aligns with the `<button>` from the a11y fix.

File: .squad/agents/wash/history.md [hunk 2]
```diff
@@ -904,3 +904,46 @@ Prevents layout drift when one card context is updated but not the other. Single
...
+- Reused existing modal pattern (backdrop click to close, stopPropagation on inner div, close button, z-50)
+- Used `downloadFile()` from space-api.ts to fetch blobs, created object URLs for media, read as text for text-like files
+- Object URLs revoked in `closeFilePreview()` to prevent memory leaks
+- Loading spinner + error-with-download-fallback pattern for async preview loading
+- File items get `cursor-pointer hover:text-slate-100` only when previewable (non-previewable files unchanged)
+
+**Learnings:**
+- `file-icons.ts` lives in `src/lib/file-icons.ts` (not in space-view directory as originally thought)
+- The download endpoint returns `application/octet- use `blob.text()` for text previews, `URL.createObjectURL(blob)` for mediastream` 
+- Pre-existing test file TS errors exist in the  only source file errors matter for validationrepo
```
[the remark this answers is not in the document]
Comment (maraf) [reply .squad/agents/wash/history.md]: ✅ Fixed in `a70d6ea`. Fully rewrote the garbled section — date corrected, filenames backticked, merged words fixed (`constantsts` → `constants`, `octet-stream` restored, etc.).

File: src/SharedSpaces.Client/e2e/screenshots.spec.ts [hunk 3]
```diff
@@ -226,6 +263,44 @@ test.describe('Screenshot Capture', () => {
...
+      await injectTokens(page, tokenMap);
+      await page.reload();
+      await page.waitForSelector('app-shell');
+      await page.click('nav button:first-child');
+      await page.waitForSelector('space-view');
+      await page.waitForTimeout(1000);
+
+      // Click the JSON file item to open text preview
+      await page.locator('p.cursor-pointer:has-text("data.json")').click();
+      await page.waitForSelector('button[aria-label="Close preview"]', { timeout: 5_000 });
```
[the remark this answers is not in the document]
Comment (maraf) [reply src/SharedSpaces.Client/e2e/screenshots.spec.ts]: ✅ Fixed in `1bc2def`. Same fix — now uses `page.getByRole('button', { name: 'data.json' })`.

File: src/SharedSpaces.Client/src/features/space-view/file-preview.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+const TEXT_EXTENSIONS = new Set([
+  // Plain text
+  'txt', 'log', 'csv', 'ini', 'conf',
+  // Structured data
+  'json', 'xml', 'yaml', 'yml', 'toml',
+  // Markdown
+  'md',
+  // Code
+  'js', 'ts', 'jsx', 'tsx', 'py', 'java', 'cs', 'go', 'rs', 'rb', 'php', 'swift', 'kt',
+  'c', 'cpp', 'h', 'hpp', 'css', 'scss', 'sass', 'less', 'html', 'sh', 'bash', 'zsh', 'sql',
```
[the remark this answers is not in the document]
Comment (maraf) [reply src/SharedSpaces.Client/src/features/space-view/file-preview.ts]: ✅ Fixed in `1bc2def`. Added `font-mono` to the text preview element — code files now render in monospace.

File: src/SharedSpaces.Client/src/features/space-view/space-view.ts [hunk 5]
```diff
@@ -1470,6 +1566,133 @@ export class SpaceView extends BaseElement {
...
+            class="h-[60vh] w-full rounded border border-slate-700"
+            title=${this.filePreviewItem?.content ?? 'PDF preview'}
+          ></iframe>
+        `;
+
+      case 'text':
+        return html`
+          <p class="whitespace-pre-wrap break-words text-start text-sm text-slate-200">
+            ${this.filePreviewText}
+          </p>
```
[the remark this answers is not in the document]
Comment (maraf) [reply src/SharedSpaces.Client/src/features/space-view/space-view.ts]: ✅ Fixed in `1bc2def`. Collapsed the `<p>` so `${this.filePreviewText}` sits directly against the tags — no phantom whitespace.
